framework/src/security/security_utils.c

```diff
@@ -124,46 +123,46 @@ hal_hmac_type secutils_convert_hmacmode_s2h(security_hmac_mode mode)
 	return HAL_HMAC_UNKNOWN;
 }
 
-security_error secutils_convert_error_h2s(hal_result_e herr)
+security_error secutils_convert_error_h2s(int herr)
 {
 	switch (herr) {
-	case HAL_SUCCESS:
+	case SECLINK_OK:
 		return SECURITY_OK;
-	case HAL_NOT_INITIALIZED:
-		return SECURITY_ERROR;
-	case HAL_INVALID_ARGS:
+	case SECLINK_INVALID_ARGS:
 		return SECURITY_INVALID_INPUT_PARAMS;
-	case HAL_INVALID_SLOT_RANGE:
+	case SECLINK_INVALID_REQUEST:
+		return SECURITY_INVALID_REQUEST;
+	case SECLINK_INVALID_SLOT_RANGE:
 		return SECURITY_ERROR;
-	case HAL_INVALID_SLOT_TYPE:
+	case SECLINK_INVALID_SLOT_TYPE:
 		return SECURITY_ERROR;
-	case HAL_EMPTY_SLOT:
+	case SECLINK_EMPTY_SLOT:
 		return SECURITY_ERROR;
-	case HAL_BAD_KEY:
+	case SECLINK_BAD_KEY:
 		return SECURITY_INVALID_KEY_INDEX;
-	case HAL_BAD_KEY_PAIR:
+	case SECLINK_BAD_KEY_PAIR:
 		return SECURITY_ERROR;
-	case HAL_BAD_CERT:
+	case SECLINK_BAD_CERT:
 		return SECURITY_INVALID_CERT_INDEX;
-	case HAL_BAD_CERTKEY_PAIR:
+	case SECLINK_BAD_CERTKEY_PAIR:
 		return SECURITY_ERROR;
-	case HAL_NOT_ENOUGH_MEMORY:
+	case SECLINK_NOT_ENOUGH_MEMORY:
```

Comment:
For memory allocation, only the reason for failure is not enough memory.
Is there any special reason to distinguish `SECLINK_NOT_ENOUGH_MEMORY` and `SECLINK_ALLOC_FAIL`?

Reply:
I think it's duplicated. I'll fix it another PR.